### ISSUE TITLE
[BUG] Unable to load columns from older versions of SQL Server.

### ISSUE THREAD

**Issue by coverrt** [cweijan/vscode-mysql#884]
When expanding the column view on a table, an SQL incorrect syntax error is thrown.

Error message:
`Error: Incorrect syntax near the keyword 'is'.`

```sql
SELECT
    [c].[COLUMN_NAME] AS [name],
    [DATA_TYPE] AS [simpleType],
    IIF
    (
        [CHARACTER_MAXIMUM_LENGTH] IS NOT NULL,
        CONCAT( [DATA_TYPE], '(', [CHARACTER_MAXIMUM_LENGTH], ')' ),
        [DATA_TYPE]
    ) AS [type],
    [IS_NULLABLE] AS [nullable],
    [CHARACTER_MAXIMUM_LENGTH] AS [maxLength],
    [COLUMN_DEFAULT] AS [defaultValue],
    '' AS [comment],
    [tc].[constraint_type] AS [key],
    COLUMNPROPERTY
    (
        OBJECT_ID( 'dbo.Acct' ),
        [c].[COLUMN_NAME],
        'IsIdentity'
    ) AS [is_identity]
FROM
    [pawnmaster].[INFORMATION_SCHEMA].[COLUMNS] AS [c]
LEFT JOIN
    [INFORMATION_SCHEMA].[CONSTRAINT_COLUMN_USAGE] AS [ccu]
        ON
            [c].[COLUMN_NAME] = [ccu].[column_name]
            AND
            [c].[table_name] = [ccu].[table_name]
            AND
            [ccu].[TABLE_SCHEMA] = [c].[TABLE_SCHEMA]
LEFT JOIN
    [INFORMATION_SCHEMA].[TABLE_CONSTRAINTS] AS [tc]
        ON
            [tc].[constraint_name] = [ccu].[constraint_name]
            AND
            [tc].[TABLE_SCHEMA] = [c].[TABLE_SCHEMA]
            AND
            [tc].[table_name] = [c].[table_name]
WHERE
    [c].[TABLE_SCHEMA] = 'dbo'
    AND
    [c].[table_name] = 'table_name'
ORDER BY
    [ORDINAL_POSITION]
```

---

The query below may be a potential fix to allow backwards compatibility with MSSQL 2008 R2 editions.

```sql
SELECT
    [c].[COLUMN_NAME] AS [name],
    [c].[DATA_TYPE] AS [simpleType],
    CASE
        WHEN [CHARACTER_MAXIMUM_LENGTH] IS NOT NULL
            THEN ([c].[DATA_TYPE] + '(' + CONVERT(NVARCHAR, [c].[CHARACTER_MAXIMUM_LENGTH]) + ')')
        ELSE
            [c].[DATA_TYPE]
    END AS [type],
    [c].[IS_NULLABLE] AS [nullable],
    [c].[CHARACTER_MAXIMUM_LENGTH] AS [maxLength],
    [c].[COLUMN_DEFAULT] AS [defaultValue],
    '' AS [comment],
    [tc].[constraint_type] AS [key],
    COLUMNPROPERTY
    (
        OBJECT_ID( 'dbo.table_name' ),
        [c].[COLUMN_NAME],
        'IsIdentity'
    ) AS [is_identity]
FROM
    [pawnmaster].[INFORMATION_SCHEMA].[COLUMNS] AS [c]
LEFT JOIN
    [INFORMATION_SCHEMA].[CONSTRAINT_COLUMN_USAGE] AS [ccu]
        ON
            [c].[COLUMN_NAME] = [ccu].[column_name]
            AND
            [c].[table_name] = [ccu].[table_name]
            AND
            [ccu].[TABLE_SCHEMA] = [c].[TABLE_SCHEMA]
LEFT JOIN
    [INFORMATION_SCHEMA].[TABLE_CONSTRAINTS] AS [tc]
        ON
            [tc].[constraint_name] = [ccu].[constraint_name]
            AND
            [tc].[TABLE_SCHEMA] = [c].[TABLE_SCHEMA]
            AND
            [tc].[table_name] = [c].[table_name]
WHERE
    [c].[TABLE_SCHEMA] = 'dbo'
    AND
    [c].[table_name] = 'table_name'
ORDER BY
    [c].[ORDINAL_POSITION]
```

---

- Operating system: Windows 11
- Database type and version: Microsoft SQL Server 2008 R2

**Comment by cweijan** [comment]
Thanks a lot! I'll fix it with a new version soon.

**Comment by cweijan** [comment]
Fixed in version 6.6.2.
